utils.c

```diff
@@ -230,6 +230,17 @@ has_prefix (const char *str,
   return strncmp (str, prefix, strlen (prefix)) == 0;
 }
 
+void
+xclearenv (void)
+{
+#ifdef HAVE_CLEARENV
```

Comment:
This project is Linux specific, so I think portability here boils down to glibc vs musl at best.  And the latter [has it it looks like](https://git.musl-libc.org/cgit/musl/tree/include/stdlib.h#n145).

But anyways, fine as is!

Reply:
I'm pleasantly surprised that musl has this, I thought they stuck to a strict position of "if it isn't guaranteed by POSIX then you don't get to use it".

If you're OK with just calling `clearenv()` unconditionally (and getting a link failure if you're using some toy libc that expects you to do `environ = NULL` like a barbarian) then I think I'd prefer that way, since I need to rebase the test anyway.

Reply:
Sounds good to me!